examples/pytorch/model_zoo/classification.py

```diff
@@ -0,0 +1,87 @@
+from dgl.data import Tox21
+from dgl.data.utils import split_dataset
+from dgl import model_zoo
+import torch
+from torch.nn import BCEWithLogitsLoss
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+
+from utils import Meter, EarlyStopping, collate_molgraphs
+
+def main():
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    batch_size = 128
+    learning_rate = 0.001
+    num_epochs = 100
+
+    # Interchangable to other Dataset
+    dataset = Tox21()
+    atom_data_field = 'h'
+
+
+    trainset, valset, testset = split_dataset(dataset, [0.8, 0.1, 0.1])
+    train_loader = DataLoader(
+        trainset, batch_size=batch_size, collate_fn=collate_molgraphs)
+    val_loader = DataLoader(
+        valset, batch_size=batch_size, collate_fn=collate_molgraphs)
+    test_loader = DataLoader(
+        testset, batch_size=batch_size, collate_fn=collate_molgraphs)
+
+    # Interchangable to other model in model zoo
+    model = model_zoo.chem.GCNClassifier(74, [64, 64], dataset.n_tasks)
+
+    model.to(device)
+    loss_criterion = BCEWithLogitsLoss(pos_weight=torch.tensor(
+        dataset.task_pos_weights).to(device), reduction='none')
+    optimizer = Adam(model.parameters(), lr=learning_rate)
+
+    stopper = EarlyStopping(patience=10)
+
+    for epoch in range(num_epochs):
+        model.train()
+        print('Start training')
+        train_meter = Meter()
+        model.train()
+        for batch_id, batch_data in enumerate(train_loader):
+            smiles, bg, labels, mask = batch_data
+            atom_feats = bg.ndata.pop(atom_data_field)
+            atom_feats, labels, mask = atom_feats.to(device), labels.to(device), mask.to(device)
+            logits = model(atom_feats, bg)
+            # Mask non-existing labels
+            loss = (loss_criterion(logits, labels)
+                    * (mask != 0).float()).mean()
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            print('epoch {:d}/{:d}, batch {:d}/{:d}, loss {:.4f}'.format(
+                epoch + 1, num_epochs, batch_id + 1, len(train_loader), loss.item()))
+            train_meter.update(logits, labels, mask)
+        train_roc_auc = train_meter.roc_auc_averaged_over_tasks()
```

Comment:
I would still suggest wrapping `L41-59` to make the main body shorter.

Reply:
I prefer not enclosing them into another function since they are short enough. The main reason is that function will take many things as argument. I would have no idea what's happening inside `run_a_train_epoch` since it's taking everything into it. Current ~80 lines I think is a proper length to read.